src/FileSystem/FilesFinder.php

```diff
@@ -55,8 +55,11 @@ private function findInDirectories(array $directories, array $suffixes): array
             ->files()
             // skip empty files
             ->size('> 0')
-            ->in($directories)
-            ->sortByName();
+            ->in($directories);
+
+        if ($sortByName) {
+            $finder->sortByName();
```

Comment:
before this PR we sorted everything everytime. this sorting is slow and should only be used when necessary